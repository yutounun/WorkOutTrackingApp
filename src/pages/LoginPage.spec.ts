import { describe, it, expect, beforeEach } from "vitest";
import { screen } from "@testing-library/dom";
import { render } from "@testing-library/vue";
import Login from "./LoginPage.vue";
import { setActivePinia, createPinia } from "pinia";

describe("Login page", () => {
  beforeEach(() => {
    // creates a fresh pinia and make it active so it's automatically picked
    setActivePinia(createPinia());
  });
  it("renders Login page properly", () => {
    render(Login);

    const loginLabel = screen.getByRole("heading", { name: "Login" });
    const emailLabel = screen.getByText("Email");
    const emailForm = screen.getByLabelText("Email");
    const passwordForm = screen.getByLabelText("Password");
    const passwordLabel = screen.getByText("Password");

    expect(loginLabel).toBeTruthy();
    expect(emailLabel).toBeTruthy();
    expect(emailForm).toBeTruthy();
    expect(passwordLabel).toBeTruthy();
    expect(passwordForm).toBeTruthy();
  });
  it("make sure nobody can see password", () => {
    render(Login);
    const passwordForm = screen.getByLabelText("Password");
    expect(passwordForm.type).toBe("password");
  });
  it("matches an old snapshot", () => {
    const Logins = render(Login);
    expect(Logins).toMatchSnapshot();
  });
});
