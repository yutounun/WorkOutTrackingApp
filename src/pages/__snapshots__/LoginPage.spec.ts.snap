// Vitest Snapshot v1

exports[`Login page > matches an old snapshot 1`] = `
{
  "baseElement": <body>
    <div>
      <main
        class="px-6 font-sans h-screen w-screen flex justify-center"
      >
        <div
          class="flex flex-col sm:justify-center lg:justify-start items-center"
        >
          <h1
            class="text-2xl lg:text-3xl pt-12 font-bold mt-5"
          >
            Login
          </h1>
          <!-- Email -->
          <div>
            <label
              class="text-xs lg:text-base font-bold text-left block mb-2 mt-16"
              for="email"
            >
               Email 
            </label>
            <input
              class="input input-bordered w-80 rounded-full w-full mb-5"
              id="email"
              placeholder="Enter Your Email"
            />
          </div>
          <div>
            <label
              class="text-xs lg:text-base font-bold text-left block mb-2 mt-5"
              for="password"
            >
               Password 
            </label>
            <input
              class="input input-bordered w-80 rounded-full w-full mb-16"
              id="password"
              placeholder="Enter Password"
              type="password"
            />
          </div>
          <!-- Login Button -->
          <div
            class="flex justify-center"
          >
            <button
              class="btn w-48 text-center text-white bg-primary hover:bg-primary mb-8"
            >
              Login
            </button>
          </div>
          <!-- Go to create account page -->
          <div
            class="text-accent"
          >
            <span>
              Don't have an account?? 
            </span>
            <a
              class="mt-5 font-semibold text-accent"
            >
               Create one here. 
            </a>
          </div>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="px-6 font-sans h-screen w-screen flex justify-center"
    >
      <div
        class="flex flex-col sm:justify-center lg:justify-start items-center"
      >
        <h1
          class="text-2xl lg:text-3xl pt-12 font-bold mt-5"
        >
          Login
        </h1>
        <!-- Email -->
        <div>
          <label
            class="text-xs lg:text-base font-bold text-left block mb-2 mt-16"
            for="email"
          >
             Email 
          </label>
          <input
            class="input input-bordered w-80 rounded-full w-full mb-5"
            id="email"
            placeholder="Enter Your Email"
          />
        </div>
        <div>
          <label
            class="text-xs lg:text-base font-bold text-left block mb-2 mt-5"
            for="password"
          >
             Password 
          </label>
          <input
            class="input input-bordered w-80 rounded-full w-full mb-16"
            id="password"
            placeholder="Enter Password"
            type="password"
          />
        </div>
        <!-- Login Button -->
        <div
          class="flex justify-center"
        >
          <button
            class="btn w-48 text-center text-white bg-primary hover:bg-primary mb-8"
          >
            Login
          </button>
        </div>
        <!-- Go to create account page -->
        <div
          class="text-accent"
        >
          <span>
            Don't have an account?? 
          </span>
          <a
            class="mt-5 font-semibold text-accent"
          >
             Create one here. 
          </a>
        </div>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "emitted": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "html": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
